upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -41,60 +42,72 @@ var _ fi.ModelBuilder = &BootstrapChannelBuilder{}
 
 // Build is responsible for adding the addons to the channel
 func (b *BootstrapChannelBuilder) Build(c *fi.ModelBuilderContext) error {
-	addons, manifests, err := b.buildManifest()
-	if err != nil {
-		return err
-	}
-
-	addonsYAML, err := utils.YamlMarshal(addons)
-	if err != nil {
-		return fmt.Errorf("error serializing addons yaml: %v", err)
-	}
-
-	name := b.cluster.ObjectMeta.Name + "-addons-bootstrap"
+	addons := b.buildAddons()
 	tasks := c.Tasks
 
-	tasks[name] = &fitasks.ManagedFile{
-		Contents:  fi.WrapResource(fi.NewBytesResource(addonsYAML)),
-		Lifecycle: b.Lifecycle,
-		Location:  fi.String("addons/bootstrap-channel.yaml"),
-		Name:      fi.String(name),
-	}
-
-	for key, manifest := range manifests {
+	for _, a := range addons.Spec.Addons {
+		key := *a.Name
+		if a.Id != "" {
+			key = key + "-" + a.Id
+		}
 		name := b.cluster.ObjectMeta.Name + "-addons-" + key
+		manifestPath := "addons/" + *a.Manifest
 
-		manifestResource := b.templates.Find(manifest)
+		manifestResource := b.templates.Find(manifestPath)
 		if manifestResource == nil {
-			return fmt.Errorf("unable to find manifest %s", manifest)
+			return fmt.Errorf("unable to find manifest %s", manifestPath)
 		}
 
 		manifestBytes, err := fi.ResourceAsBytes(manifestResource)
 		if err != nil {
-			return fmt.Errorf("error reading manifest %s: %v", manifest, err)
+			return fmt.Errorf("error reading manifest %s: %v", manifestPath, err)
 		}
 
 		manifestBytes, err = b.assetBuilder.RemapManifest(manifestBytes)
 		if err != nil {
-			return fmt.Errorf("error remapping manifest %s: %v", manifest, err)
+			return fmt.Errorf("error remapping manifest %s: %v", manifestPath, err)
+		}
+
+		rawManifest := string(manifestBytes)
+		klog.V(4).Infof("Manifest %v", rawManifest)
+
+		manifestHash, err := utils.HashString(&rawManifest)
+		klog.V(4).Infof("hash %s", manifestHash)
+		if err != nil {
+			manifestHash = ""
```

Comment:
Typically a good idea to warn if there's any error we're ignoring.  I'd probably return error here, as we should always be able to hash a string!